Refactor STM32F0xx dtsi

gautierg-st opened the pull request:
This PR reorganizes the successive dtsi inclusion of the STM32F0 series to better reflect the subfamilies (STM32F0x0 on one side, STM32F0x1, x2 and x8 on the other). It allows to fix some errors of peripherals included in devices that do not have them, and to remove some duplication by having a more logical and clear inclusion tree.

Also add support for STM32F042 and Nucleo F042K6.

gautierg-st committed:
dts: arm: st: f0: add stm32f042 dtsi

Add a dtsi for STM32F042 for the support of Nucleo F042K6

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
soc: arm: st_stm32: stm32f0: add kconfig for stm32f042x6

Add Kconfig for STMF042x6 to support the Nucleo F042K6.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: arm: nucleo_f042k6: add support for nucleo_f042k6

Add support for Nucleo F042K6

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

erwango commented on boards/arm/nucleo_f042k6/nucleo_f042k6.yaml:
```diff
@@ -0,0 +1,18 @@
+supported:
+  - i2c
+  - spi
+  - gpio
```
adc

erwango commented on boards/arm/nucleo_f042k6/nucleo_f042k6.yaml:
```diff
@@ -0,0 +1,18 @@
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 6
```
On some other devices with limited available RAM, gpio d and f where disabled.
Is it safe to keep them?
Did you run compilation campaign on this board to detect test with potential RAM overflow ?

erwango commented on dts/arm/st/f0/stm32f091.dtsi:
```diff
@@ -4,73 +4,60 @@
+		 * these may be enabled at a time. Enabling more than one will
+		 * result in a build failure.
+		 */
+		usart5: serial@40005000 {
```
This looks like an addition rather than a reorganization. A dedicated commit is required to ease the review.

erwango reviewed (changes requested):
First commit would require to be split to ease review.
This is specially required if new nodes are added.

gautierg-st committed:
dts: arm: st: f0: Refactor stm32f0xx inclusion tree

Modify the successive dtsi include to better reflect the underlying
structure of the F0 family.
There are two main subfamilies: STM32F0x0 on one side, and STM32F0x1, x2
and x8 on the other

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 usart2 driver

Remove USART2 from base dtsi, and add it to F030x8, F070 & F051

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 i2c2 driver

Remove I2C2 from base dtsi, and add it to F030x8, F070xB & F051

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 timer drivers

Remove TIM6, 7 & 15 from base dtsi, and add TIM6 & 15 to F030x8, TIM7 for
F030xC, TIM15 for F070, TIM6 & 7 for F070xC, TIM6 & 15 to F051, and TIM7
for F071.
Remove TIM2 from F072 and F091 dtsi since it is already included in F031.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 dma1 interrupt channels

Fix DMA1 interrupt channels. There are 7 for STM32F071 and higher, and 5
for all others, default value for the series.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 temperature driver

Remove calibrated temperature measure from base dtsi since it does not
exist for STM32F0x0, and add it only for the other STM32F0.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 flash erase-block-size

Flash erase-block-size is 2048 for F030xC, F070xB, and F071 and higher.
For all others, it is 1024, default value in base dtsi.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 spi2 driver

Remove SPI2 from F070 (not present on F070x6) and add it to F070xB.
Add it to F051 and remove it from F091 (since it is already defined).

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: remove duplicate of rtc backup registers

RTC backup registers are already defined in F031, so no need to include
them again.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 hsi48 driver

Add HSI48 to F071, and remove it from F072 and F091.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 gpioe driver

Add GPIOE to F071, and remove it from F072 and F091.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: refactor stm32f0 dac1 driver

Remove DAC1 from F072 and F091 since it alfready defined in F051.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: add usb to stm32f072

Add USB to STM32F072 dtsi.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: add can1 to stm32f091

Add CAN1 to STM32F091 dtsi.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: f0: add supplementary usart for stm32f0

Add some more USART for F070xB, F071 and F091.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>
